### DISCUSSION TITLE
Runs, but does not respond to mouse clicks.

### DISCUSSION THREAD

**Discussion by robertflesch** [ocornut/imgui#7667] [New Users Build/Link/Run issues ONLY!]
New to ImGui, but long time UI developer.
Running on Vulkan 1.3, latest ImGui, Windows 11, VS22

Running some sample code from ImGui FAQ.
It runs beautifully, I can see my mouse overs, but none of the controls responds to mouse clicks.

https://github.com/ocornut/imgui/assets/1186354/6244fddf-34a9-4dc9-9add-f23cf6464a5a

You can kind of see in demo I am hovering over Button, I am clicking on it. but number does not change, nor do any of the other controls respond. Also true for demo window. I can highlight an expandable control, but it never opens.

Looking at example code, I don't see where you are passing in the clicks from glfw, what am I missing?

while (!glfwWindowShouldClose(window)) {
    glfwPollEvents();
    
    ImGui_ImplVulkan_NewFrame();
    ImGui_ImplGlfw_NewFrame();
    ImGui::NewFrame();

    {
        bool show_demo_window = true;
        bool show_another_window = false;
		ImVec4 clear_color = ImVec4(0.45f, 0.55f, 0.60f, 1.00f);
        static float f = 0.0f;
        static int counter = 0;

        ImGui::Begin("Hello, world!");                          // Create a window called "Hello, world!" and append into it.

        ImGui::Text("This is some useful text.");               // Display some text (you can use a format strings too)
        ImGui::Checkbox("Demo Window", &show_demo_window);      // Edit bools storing our window open/close state
        ImGui::Checkbox("Another Window", &show_another_window);

        ImGui::SliderFloat("float", &f, 0.0f, 1.0f);            // Edit 1 float using a slider from 0.0f to 1.0f
        ImGui::ColorEdit3("clear color", (float*)&clear_color); // Edit 3 floats representing a color

        if (ImGui::Button("Button"))                            // Buttons return true when clicked (most widgets return true when edited/activated)
            counter++;
        ImGui::SameLine();
        ImGui::Text("counter = %d", counter);
        ImGui::End();
    }

    ImGui::Render();
   ImGui::EndFrame();
}

**Comment by ocornut** [comment]
The ImGui_ImplGlfw_InitXXX functions is installing callbacks or you can install them yourself and call the callbacks functions declared in imgui_impl_glfw.h.

**Reply by robertflesch** [reply]
So I reviewed my code, checked to make sure that I was calling ImGui_ImplGlfw_InitForVulkan, and indeed I do. As well as calling ImGui_ImplVulkan_Init. I would have thought that if I was seeing the mouse move messages, I would be getting the mouse click messages as well. Any clues as to how to debug it?

void Renderer::imgui_init() {
    ImGui_ImplGlfw_InitForVulkan(this->window, false);
    ImGui_ImplVulkan_InitInfo init_info = {};
    init_info.Instance = _manager->getInstance();
    init_info.PhysicalDevice = _manager->getPhysicalDevice();
    init_info.Device = _manager->getDevice();
    init_info.Queue = _manager->getGraphicsQueue();
    init_info.DescriptorPool = imguiPool;
    init_info.MinImageCount = 3;
    init_info.ImageCount = 3;
    init_info.MSAASamples = VK_SAMPLE_COUNT_1_BIT;

    ImGui_ImplVulkan_Init(&init_info, renderPass);
}

**Reply by robertflesch** [reply]
Found the problem! 
ImGui_ImplGlfw_InitForVulkan(this->window, false); Does NOT install callbacks, changing it to true does! 
Thx for the help.
